UPGRADE-3.1.md

```diff
@@ -24,6 +24,22 @@ FrameworkBundle
 
  * As it was never an officially supported feature, the support for absolute
    template paths has been deprecated and will be removed in Symfony 4.0.
+ * Since the refactoring of 2.8, form types without dependencies are autoloaded
+   by the `DependencyInjectionExtension` in the form registry.
+   Consequently they are deprecated and will be removed in 4.0.
+   You should use the fully-qualified name instead.
+
+   Before:
+
+   ```php
+   $type = $this->container->get('form.type.text');
+   ```
+
+   After:
```

Comment:
Your before and after cases are not the same (``$type`` is an instance before and a class name after).

The only case of using the service directly before would be to customize the form type (which should be done using proper form extension points instead anyway, as it cannot be done cleanly by overwriting the service)